com.unity.sg2/Editor/GraphUI/ShaderGraphStencil.cs

```diff
@@ -135,47 +135,47 @@ protected override void CreateGraphProcessors()
-                var registry = graphModel.RegistryInstance;
-                var graphHandler = graphModel.GraphHandler;
+            var graphModel = (ShaderGraphModel)model.GraphModel;
 
-                // If the guid starts with a number, it will produce an invalid identifier in HLSL.
-                var variableDeclarationName = "_" + graphDataVar.Guid;
-                var contextName = graphModel.BlackboardContextName;
+            // Use this variables' generated guid to bind it to an underlying element in the graph data.
+            var registry = graphModel.RegistryInstance;
+            var graphHandler = graphModel.GraphHandler;
 
-                var propertyContext = graphHandler.GetNode(contextName);
-                Debug.Assert(propertyContext != null, "Material property context was missing from graph when initializing a variable declaration");
+            // If the guid starts with a number, it will produce an invalid identifier in HLSL.
+            var variableDeclarationName = "_" + graphDataVar.Guid;
+            var contextName = graphModel.BlackboardContextName;
 
-                var entry = new ContextEntry
-                {
-                    fieldName = variableDeclarationName,
-                    height = ShaderGraphExampleTypes.GetGraphTypeHeight(model.DataType),
-                    length = ShaderGraphExampleTypes.GetGraphTypeLength(model.DataType),
-                    primitive = ShaderGraphExampleTypes.GetGraphTypePrimitive(model.DataType),
-                    precision = GraphType.Precision.Any,
-                    initialValue = Matrix4x4.zero,
-                };
-
-                ContextBuilder.AddReferableEntry(propertyContext, entry, registry.Registry, ContextEntryEnumTags.PropertyBlockUsage.Included, displayName: variableDeclarationName);
-                try
-                {
-                    graphHandler.ReconcretizeNode(propertyContext.ID.FullPath);
-                }
-                catch (Exception e)
-                {
-                    Debug.LogException(e);
-                }
+            var propertyContext = graphHandler.GetNode(contextName);
+            Debug.Assert(propertyContext != null, "Material property context was missing from graph when initializing a variable declaration");
 
-                graphDataVar.contextNodeName = contextName;
-                graphDataVar.graphDataName = variableDeclarationName;
+            var entry = new ContextEntry
+            {
+                fieldName = variableDeclarationName,
+                height = ShaderGraphExampleTypes.GetGraphTypeHeight(model.DataType),
+                length = ShaderGraphExampleTypes.GetGraphTypeLength(model.DataType),
+                primitive = ShaderGraphExampleTypes.GetGraphTypePrimitive(model.DataType),
+                precision = GraphType.Precision.Any,
+                initialValue = Matrix4x4.zero,
+            };
```

Comment:
Just a heads up, there are probably going to be some conflicts here with with what Joe is working on.